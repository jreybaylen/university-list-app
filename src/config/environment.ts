const environment = {
    delimiter: '$@',
    userStorage: 'univ-app-user',
    authStorage: 'univ-app-user-auth',
    saveUnivStorage: 'univ-app-user-universities'
}

export {
    environment
}