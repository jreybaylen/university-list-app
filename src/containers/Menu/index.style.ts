import { StypesProps } from '@interface/style.interface'

const styles: StypesProps = {
    container: {
        width: '100%',
        maxWidth: 1000,
        margin: 'auto',
        paddingTop: 20,
        display: 'flex',
        justifyContent: 'center'
    },
    navContainer: {
        paddingLeft: 0,
        display: 'flex',
        listStyle: 'none',
        justifyContent: 'center'
    },
    link: {
        color: '#2196f3',
        borderRadius: 25,
        padding: '10px 20px',
        textDecoration: 'none'
    },
    activeLink: {
        background: '#f1f1f1',
        border: '1px solid #ccc'
    },
    name: {
        fontWeight: 500,
        margin: '0 15px'
    },
    signOut: {
        marginLeft: 5,
        fontWeight: 500,
        paddingLeft: 20,
        cursor: 'pointer',
        borderLeft: '2px solid #ccc'
    }
}

export {
    styles
}