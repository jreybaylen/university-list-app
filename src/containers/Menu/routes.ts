import { RoutesProps } from './index.interface'

const routes = (isAuth: boolean): RoutesProps => {
    const config = [
        {
            id: 'home',
            path: '/',
            label: 'Home'
        },
        {
            id: 'sign-in',
            path: '/auth',
            label: 'Sign In'
        },
        {
            id: 'register',
            path: '/register',
            label: 'Register'
        },
    ]
    const modifiedConfig = config.slice(0, isAuth ? 1 : config.length)

    return modifiedConfig
}

export {
    routes
}