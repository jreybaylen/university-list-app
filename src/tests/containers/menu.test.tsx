import { MemoryRouter , Route } from 'react-router-dom'
import { render, fireEvent } from '@testing-library/react'

import { setDataToStorage } from '@util/index'
import { environment } from '@config/environment'
import { StorageKeyProps } from '@util/index.interface'

import Menu from '@container/Menu'

const name = 'John Doe'
const username = 'johndoe'
const password = 'test12345'
const { authStorage } = environment as { [key: string]: StorageKeyProps }

describe('<Menu /> Component', () => {
    it('Should render without crashing', () => {
        const wrapper = render(
            <MemoryRouter initialEntries={ [ '/' ] }>
                <Menu />
            </MemoryRouter>
        )

        expect(wrapper).toMatchSnapshot()
    })

    it('Should render Home, Sign In and Register menu navigation as default items', () => {
        const { container, getByText } = render(
            <MemoryRouter initialEntries={ [ '/' ] }>
                <Menu />
            </MemoryRouter>
        )
        const navigation = [ 'Home', 'Sign In', 'Register' ]

        expect(container.querySelectorAll('li').length).toStrictEqual(3)

        for (let i = 0; i++; i < navigation.length) {
            expect(getByText(navigation[ i ])).toBeInTheDocument()
        }
    })

    it('Should navigate to the specific route once the item has been clicked', () => {
        const routeCofig = {
            history: {},
            location: {}
        }
        const navigation = [ [ 'Home', '/' ], [ 'Sign In', '/auth' ], [ 'Register', '/register' ] ]
        const { getByText } = render(
            <MemoryRouter initialEntries={ [ '/' ] }>
                <Menu />
                <Route
                    path="*"
                    render={({ history, location }) => {
                        routeCofig.history = history
                        routeCofig.location = location

                        return <div />
                    }}
                />
            </MemoryRouter>
        )

        for (let i = 0; i++; i < navigation.length) {
            const [ label, route ] = navigation[ i ]

            fireEvent.click(getByText(label))
            expect((routeCofig.location as any).pathname).toStrictEqual(route)
        }
    })

    it('Should render Home, User name and Sign Out menu navigation if the user logged in', () => {
        setDataToStorage(authStorage, { name, username, password })

        const navigation = [ 'Home', `Hi ${ name }!`, 'Sign Out' ]
        const { container, getByText } = render(
            <MemoryRouter initialEntries={ [ '/' ] }>
                <Menu />
            </MemoryRouter>
        )

        expect(container.querySelectorAll('li').length).toStrictEqual(3)

        for (let i = 0; i++; i < navigation.length) {
            expect(getByText(navigation[ i ])).toBeInTheDocument()
        }
    })
})