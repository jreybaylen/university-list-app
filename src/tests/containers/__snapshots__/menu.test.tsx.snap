// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Component Should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        style="width: 100%; max-width: 1000px; margin: auto; padding-top: 20px; display: flex; justify-content: center;"
      >
        <ul
          style="padding-left: 0px; display: flex; list-style: none; justify-content: center;"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
              style="color: rgb(33, 150, 243); border-radius: 25px; padding: 10px 20px; text-decoration: none; text-transform: capitalize; font-weight: 500; background: rgb(241, 241, 241); border: 1px solid #ccc;"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/auth"
              style="color: rgb(33, 150, 243); border-radius: 25px; padding: 10px 20px; text-decoration: none; text-transform: capitalize;"
            >
              Sign In
            </a>
          </li>
          <li>
            <a
              href="/register"
              style="color: rgb(33, 150, 243); border-radius: 25px; padding: 10px 20px; text-decoration: none; text-transform: capitalize;"
            >
              Register
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      style="width: 100%; max-width: 1000px; margin: auto; padding-top: 20px; display: flex; justify-content: center;"
    >
      <ul
        style="padding-left: 0px; display: flex; list-style: none; justify-content: center;"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            style="color: rgb(33, 150, 243); border-radius: 25px; padding: 10px 20px; text-decoration: none; text-transform: capitalize; font-weight: 500; background: rgb(241, 241, 241); border: 1px solid #ccc;"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/auth"
            style="color: rgb(33, 150, 243); border-radius: 25px; padding: 10px 20px; text-decoration: none; text-transform: capitalize;"
          >
            Sign In
          </a>
        </li>
        <li>
          <a
            href="/register"
            style="color: rgb(33, 150, 243); border-radius: 25px; padding: 10px 20px; text-decoration: none; text-transform: capitalize;"
          >
            Register
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
